### PR TITLE
updated web3 and deployer logic

### PR DESCRIPTION
**Pull request by BatiGencho**

### CODE REVIEW

File: packages/sc-deployer/src/index.ts [hunk 1]
```diff
@@ -224,28 +302,26 @@ async function deploySC(
 
   console.log('Waiting for events...');
 
-  // await finalization if required
-  await awaitOperationFinalization(client, opId);
+  // async poll events in the background for the given opId
+  const { isError, eventPoller, events }: IEventPollerResult =
+    await time.withTimeoutRejection<IEventPollerResult>(
```
Comment (peterjah) [inline packages/sc-deployer/src/index.ts:307]: Not directly related to this PR, but its a shame that we reimplemented theses time related function.. everything already exists somewhere in an npm package and we should not reinvent the wheel. this "time" object has nothing to do in our web3 repo...
wait function: https://www.npmjs.com/package/delay
set Timeout: https://www.npmjs.com/package/await-timeout
etc...

**Review by peterjah — APPROVED** [review]

File: packages/sc-deployer/src/index.ts (hunk 1 above)
Comment (BatiGencho) [reply packages/sc-deployer/src/index.ts:307]: Do you insist on using the delay and await-timeout packages ? Can I do that in a separate MR ?

**Review by Thykof — CHANGES_REQUESTED** [review]
contract.coins.toValue is not a function

**Comment by Thykof** [comment]
I ran:

```
 npm update @massalabs/as-transformer @massalabs/as-types @massalabs/eslint-config @massalabs/massa-as-sdk @massalabs/massa-sc-compiler @massalabs/massa-sc-deployer @massalabs/massa-web3
npm run build
npm i
npx sc-project-initializer init ~/dev/test-toolkit/testDir9
```

and in `~/dev/test-toolkit/testDir9`

I run: `npm run deploy`

I am having the error: 

```
2 files to compile
contract to compile assembly/contracts/main.ts

contract to compile assembly/contracts/run.ts

Wallet Address: A1qYB7FWXHs2Ydm7HG4wBNpzpzfDhjWBj7PV1Rsg2MoqbZTHPHr has balance (candidate, final) = (1000000000, 1000000000) MAS
TypeError: contract.coins.toValue is not a function
    at file:///Users/nathanmassa/dev/test-toolkit/testDir9/node_modules/@massalabs/massa-sc-deployer/src/index.ts:201:45
    at Array.reduce (<anonymous>)
    at deploySC (file:///Users/nathanmassa/dev/test-toolkit/testDir9/node_modules/@massalabs/massa-sc-deployer/src/index.ts:200:27)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///Users/nathanmassa/dev/test-toolkit/testDir9/src/deploy.ts:26:3
```

File: packages/sc-deployer/src/index.ts (hunk 1 above)
Comment (Thykof) [reply packages/sc-deployer/src/index.ts:307]: he approve so you should create an issue to to in later

**Comment by Thykof** [comment]
My bad I did npm link the deployer, however make sure to create a following PR to update the dependencies in command/init

**Comment by Thykof** [comment]
now I am getting Timeout of 20000 has passed and promise did not resolve.

It could be great to make it work the first try as describe in the documentation https://docs.massa.net/en/latest/web3-dev/smart-contracts/getting-started.html

**Comment by BatiGencho** [comment]
@Thykof Please try now with v.15.1 from web3

**Comment by Thykof** [comment]
So to make it work against the testnet we need to provide 0.1 as fees in the deployment script.

**Review by Thykof — APPROVED** [review]

**Review by Thykof — APPROVED** [review]

**Review by Thykof — APPROVED** [review]

**Pull request merged**